### PR TITLE
chore(kno-1566): add tenant interface

### PR DESCRIPTION
**Pull request by meryldakin**
Formats with rubocop and adds tenant methods.

### CODE REVIEW

**Comment by meryldakin** [comment]
I ran the rubocop formatter here which is all of the other file changes but let me know if there's a reason we might not want to run that for this repo!

File: lib/knock/tenants.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+require 'uri'
+
+module Knock
+  # Methods for interacting with tenants in Knock
+  module Tenants
+    class << self
+      include Base
+      include Client
+
+      DEFAULT_PREFERENCE_SET_ID = 'default'
```
Comment (cjbell) [inline lib/knock/tenants.rb:13]: Nit: this is unused

File: lib/knock/tenants.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+      # Retrieves all Tenants in environment
+      #
+      # @return [Hash] Paginated list of Tenant records
+      def list
```
Comment (cjbell) [inline lib/knock/tenants.rb:18]: We should be passing options to this endpoint for filters and page size and such?

**Review by cjbell — COMMENTED** [review]
Two comments but looking great 💪 thanks for getting tests in!

**Review by cjbell — COMMENTED** [review]
💯💯💯

**Review by cjbell — APPROVED** [review]

**Pull request merged**